### DISCUSSION TITLE
Best way to invoke from a python program

### DISCUSSION THREAD

**Discussion by answerquest** [MobilityData/gtfs-validator#1665] [Q&A] [answered]
Hi, big thanks to the developers and hosters of this tool, it's a big improvement over the old google feedvalidator.

I have a python backend deployed on a server, where, as part of the gtfs output flow I want it to be able to "call" this program and get the report.json it produces in return. Now the best thing would be if this was available as an api call. But I understand that's not the makeup of the program as it currently is. (but this could be a worthy api-wrapper project, tip for those looking for contribution opportunities)

Still, I'd like to know which ways have been tried out by folks.
I'm thinking of including the jar file into my dockerized deployment. But my backend is in python. My base image is: `python:3.12.1-slim-bookworm`. From python backend I'm able to run shell commands to outsource work and then pickup the results. 

Would anybody know what to add to Dockerfile to ensure that the .jar works inside the docker container?

**Comment by answerquest** [comment, marked as answer]
Got it! Have to add this line in my dockerfile:
```
RUN apt update && apt install -y default-jre 
```
> Note: This is valid for my debian OS based docker image `python:3.12.1-slim-bookworm`. There may be other commands for yours if you're using alpine etc. Please lookup online for "how to install java runtime environment in [your docker OS]".   
You can find out by doing a docker build with just `RUN apt update` and seeing where the libs etc are being fetched from. In my case it showed debian. Else look up the publisher of your base docker image.


I included the .jar in my code folder, it's sitting next to my python files.
Once this is baked in, I'm able to successfully run this shell command from my python backend code:
```
java -jar gtfs-validator-4.2.0-cli.jar -o "/home/output" -i "/home/output/gtfs.zip" -r "report.html"
```
here, a persistent volume with rw permissions is mounted on /home/output/ path, and this ensures that the java program has the necessary permissions to run and write the output files here.

Once the shell process has completed, I'm picking up the html and json files created in the output folder for next steps.

**Comment by davidgamez** [comment]
Hi @answerquest, welcome aboard! I'm glad that you found a solution, and the community really appreciates that you shared it back with a detailed explanation. Thanks!
